packages/clay-core/src/table/Body.tsx

```diff
@@ -20,14 +21,83 @@ type Props<T> = {
 	items?: Array<T>;
 } & React.TableHTMLAttributes<HTMLTableSectionElement>;
 
+type ItemProps<T> = {
+	children: React.ReactElement;
+	item: T;
+	keyValue: React.Key;
+};
+
+// TODO: Move the implementation to the Collection and use the benefits of
+// generator to pause the iteration and return later with a form of
+// yield CPU time for the browser.
+function* flatten<T extends Record<string, any>>(
+	array: Array<T>,
+	expandedKeys: Set<React.Key>,
+	nestedKey: string,
+	level = 1
+): IterableIterator<T> {
+	for (let i = 0; i < array.length; i++) {
+		const item = {
+			...array[i],
+			_index: i,
+			_level: level,
+			_size: array.length,
+		} as unknown as T;
+
+		if (Array.isArray(array[i]![nestedKey])) {
+			delete item[nestedKey];
+			// @ts-ignore
+			item._expandable = true;
+			yield item;
+
+			// TODO: Make it dynamic using the key generated by the developer at
+			// render time.
+			if (expandedKeys.has(array[i]!['id'])) {
```

Comment:
I spent some time trying to make this data agnostic for `<Row />` to get the key through the collection so that the developer can configure it regardless of their data structure but I had problems with this because the flatten it must be done before passing the data to the collection for rendering and this was a problem and this way I cannot obtain this information because this happens before and at the time of rendering.

When we move this to be part of the collection API perhaps we can obtain this dynamically, meaning that the layout creation is still in a tree so it would be done before and the flattening would happen at render time. TreeView can benefit from this as well.